src/ConnectedMode/Migration/FileProviders/XmlFileProvider.cs

```diff
@@ -87,8 +87,13 @@ public async Task<IEnumerable<string>> GetFilesAsync(CancellationToken token)
             }
 
             var allFiles = new HashSet<string>(roslynProjectFiles, StringComparer.OrdinalIgnoreCase);
-            var foundFiles = await GetFilesFromFileSystemAsync();
-            AddToMatches(allFiles, foundFiles);
+
+            // Make sure we are doing the disk search on a background thread
+            await threadHandling.SwitchToBackgroundThread();
+
+            await AddFilesUnderSolutionFolderAsync(allFiles);
+            AddFilesInProjectFolders(allFiles, roslynProjectFiles);
```

Comment:
I re-factored the existing code to make it easier to see the two logical searchers that are being carried out: under the solution folder, and under each project folder.